File: modules/expo-mlkit-core/ios/ExpoMLKitRect.swift
import ExpoModulesCore

public struct ExpoMLKitRect: Record {
    public init(){}
    
    @Field
    var origin: ExpoMLKitPoint = ExpoMLKitPoint()
    @Field
    var size: ExpoMLKitPoint = ExpoMLKitPoint()

    public static func fromCGRect(rect: CGRect) -> ExpoMLKitRect {
        let expoRect = ExpoMLKitRect()

        let origin = ExpoMLKitPoint()
        origin.x = rect.origin.x
        origin.y = rect.origin.y
        
        let size = ExpoMLKitPoint()
        size.x = rect.width
        size.y = rect.height
        
        expoRect.origin = origin;
        expoRect.size = size
        
        return expoRect
    }
}

File: modules/expo-mlkit-core/android/src/main/java/red/infinite/expomlkit/core/ExpoMLKitRect.kt
package red.infinite.expomlkit.core

import android.graphics.Rect
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

data class ExpoMLKitRect(
    @Field var origin: ExpoMLKitPoint,
    @Field var size: ExpoMLKitPoint
) : Record {
    companion object {
        fun fromRect(rect: Rect): ExpoMLKitRect {
            return ExpoMLKitRect(
                origin = ExpoMLKitPoint(rect.left.toFloat(), rect.top.toFloat()),
                size = ExpoMLKitPoint(rect.width().toFloat(), rect.height().toFloat())
            )
        }

        fun zero(): ExpoMLKitRect {
            return ExpoMLKitRect(
                origin = ExpoMLKitPoint.zero(),
                size = ExpoMLKitPoint.zero()
            )
        }
    }


}